src/main/java/com/minecolonies/coremod/generation/SifterCraftingProvider.java

```diff
@@ -0,0 +1,364 @@
+import net.minecraft.data.IDataProvider;
+import net.minecraft.data.IFinishedRecipe;
+import net.minecraft.item.Item;
+import net.minecraft.item.ItemStack;
+import net.minecraft.item.Items;
+import net.minecraft.util.ResourceLocation;
+import net.minecraftforge.registries.ForgeRegistries;
+import org.jetbrains.annotations.NotNull;
+
+import java.io.IOException;
+import java.nio.file.Path;
+import java.util.*;
+import java.util.function.Consumer;
+import java.util.stream.Collectors;
+import java.util.stream.Stream;
+
+import static com.minecolonies.api.util.constant.Constants.MOD_ID;
+
+public class SifterCraftingProvider implements IDataProvider
+{
+    private final SifterRecipeProvider recipeProvider;
+    private final SifterLootTableProvider lootTableProvider;
+    private final Map<Item, List<SifterMeshDetails>> inputs = new HashMap<>();
+
+    public SifterCraftingProvider(@NotNull final DataGenerator generatorIn)
+    {
+        inputs.put(Items.DIRT, Stream.of(
+                new SifterMeshDetails(ModItems.sifterMeshString, 1, new LootTableBuilder()
+                        .empty(71)
+                        .item(Items.WHEAT_SEEDS, 25)
+                        .item(Items.OAK_SAPLING, 1)
+                        .item(Items.BIRCH_SAPLING, 1)
+                        .item(Items.SPRUCE_SAPLING, 1)
+                        .item(Items.JUNGLE_SAPLING, 1)
+                        .build()),
+
+                new SifterMeshDetails(ModItems.sifterMeshFlint, 3, new LootTableBuilder()
+                        .empty(28)
+                        .item(Items.WHEAT_SEEDS, 50)
+                        .item(Items.OAK_SAPLING, 5)
+                        .item(Items.BIRCH_SAPLING, 5)
+                        .item(Items.SPRUCE_SAPLING, 5)
+                        .item(Items.JUNGLE_SAPLING, 5)
+                        .item(Items.CARROT, 1)
+                        .item(Items.POTATO, 1)
+                        .build()),
+
+                new SifterMeshDetails(ModItems.sifterMeshIron, 4, new LootTableBuilder()
+                        .bonusRolls(0.025f)
+                        .empty(3)
+                        .item(Items.WHEAT_SEEDS, 50)
+                        .item(Items.OAK_SAPLING, 10)
+                        .item(Items.BIRCH_SAPLING, 10)
+                        .item(Items.SPRUCE_SAPLING, 10)
+                        .item(Items.JUNGLE_SAPLING, 10)
+                        .item(Items.CARROT, 1)
+                        .item(Items.POTATO, 1)
+                        .item(Items.PUMPKIN_SEEDS, 1)
+                        .item(Items.MELON_SEEDS, 1)
+                        .item(Items.BEETROOT_SEEDS, 1)
+                        .item(Items.DARK_OAK_SAPLING, 1)
+                        .item(Items.ACACIA_SAPLING, 1)
+                        .build()),
+
+                new SifterMeshDetails(ModItems.sifterMeshDiamond, 5, new LootTableBuilder()
+                        .bonusRolls(0.025f)

```

Comment:
I've kept this value from the original loot table, but I suspect that it's a typo, since all of the other diamond mesh tables have a bonus roll of 0.035 instead.

Reply:
Yes, it's a typo.

Reply:
Ok, I'll fix that.

Reply:
Fixed.